lib/ansible/modules/utilities/logic/include_vars.py

```diff
@@ -92,6 +92,21 @@
       - "{{ ansible_os_family }}.yaml"
       - default.yaml
 
+- name: Additive loading of OS specific variables
+  include_vars: "{{ vars_item }}"
+  when: vars_item | length
+  vars:
+      vars_item: "{{ lookup('first_found', vars_search) }}"
+      vars_search:
+          files: "{{ item }}"
+          paths: vars
+          skip: true
+  loop:
+      - "{{ ansible_os_family | lower }}.yml"
+      - "{{ ansible_distribution | lower }}.yml"
+      - "{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
+      - "{{ ansible_distribution | lower }}-{{ ansible_distribution_version.replace('.', '-') | lower }}.yml"
+
```

Comment:
@bcoca Kinda off, mainly I made a mistake while trying to simplify original example from https://github.com/ansible/ansible/issues/38227 

The with first_found part can probably be remove to avoid two loops, but the the big loop is a must, because this is what we want to cover: loading all files that do match in that granularity order so you could add a variable in a single file if you want, or override it in a more-granular one.

I will wait for your oppinion if I should just include the original example which is more complex or redo this one and drop the feature that made the other one more complex (lookin in various locations for these files).

Anyway the highlighted piece of code is the correct one.

Reply:
ah, ok, so this is 'load all found' ... the 'first found' usage threw me off. still there is a better way:
```
    when: lookup('pipe', '[ -f ' ~ item ~ ' ]; echo $?') == '0'
    loop:
        - "{{ ansible_os_family | lower }}.yml"
        - "{{ ansible_distribution | lower }}.yml"
        - "{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
        - "{{ ansible_distribution | lower }}-{{ ansible_distribution_version.replace('.', '-') | lower }}.yml"
```

Reply:
even cleaner
```yaml
  - include_vars: "{{ item }}"
    when: item is file
    loop:
        - "{{ ansible_os_family | lower }}.yml"
        - "{{ ansible_distribution | lower }}.yml"
        - "{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
        - "{{ ansible_distribution | lower }}-{{ ansible_distribution_version.replace('.', '-') | lower }}.yml"
```

Reply:
@bcoca just wondering how to get your "cleaner" example working when files exist in vars directory.   For me at least neither option you mentioned seemed to find the files in the vars directory.   I had to switch to the below option to get it to work.  I don't like the option I am currently using as I believe 
a) it does not not show a failure if the syntax of the included vars file has errors
b) it doesn't show which file was loaded, in the other options that you mention it does show skipped when the file doesn't exist

I much prefer your cleaner options but it doesn't seem to pass the "when" check when the files exist in the vars directory.
```
- name: "load os specific vars"
  include_vars: "{{ item }}"
  when: item is file
  failed_when: false
  loop:
    - "{{ ansible_os_family | lower }}.yml"
    - "{{ ansible_distribution | lower }}.yml"
    - "{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
    - "{{ ansible_distribution | lower }}-{{ ansible_distribution_version.replace('.', '-') | lower }}.yml"
```

Reply:
you have to add 'vars/' to the find then, my option is not meant to work in all the same cases, but it is cleaner (IMHO) for the case presented.

Reply:
not sure if this is what you meant, but I did also try this which didn't work:
```
- include_vars: "{{ item }}"
  when: item is file
  loop:
    - "vars/{{ ansible_os_family | lower }}.yml"
    - "vars/{{ ansible_distribution | lower }}.yml"
    - "vars/{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
    - "vars/{{ ansible_distribution | lower }}-{{ ansible_distribution_version.replace('.', '-') | lower }}.yml"
```

The best option for me was the one presented earlier by @flowerysong which does seem to work well:
```
- name: Collect OS variables
  include_vars: "{{ vars_item }}"
  when: vars_item | length
  vars:
    vars_item: "{{ lookup('first_found', vars_search) }}"
    vars_search:
      files: "{{ item }}"
      paths:
        - defaults/os
        - defaults
        - vars/os
        - vars
      skip: true
  loop:
    - "{{ ansible_os_family | lower }}.yml"
    - "{{ ansible_distribution | lower }}.yml"
    - "{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version | lower }}.yml"
    - "{{ ansible_distribution | lower }}-{{ ansible_distribution_version.replace('.', '-') | lower }}.yml"
```